src/Audit/Parser/NpmAuditParser.php

```
@@ -0,0 +1,116 @@
+        $dependencies = $this->getObject($metadata['dependencies'] ?? null, 'metadata.dependencies');
+
+        foreach (['prod', 'dev', 'optional', 'peer', 'peerOptional', 'total'] as $dependencyType) {
+            $this->getNonNegativeInteger($dependencies, $dependencyType, 'metadata.dependencies');
+        }
+
+        if ($severityTotal !== count($vulnerabilities)) {
+            throw $this->malformed('metadata vulnerability counts must equal the vulnerability entries');
+        }
+
+        return $findings;
+    }
```

PHPMD (2.15.0)

[warning] The method parse() has a Cyclomatic Complexity of 16. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)